packages/atomic/src/utils/initialization-utils.tsx

```diff
@@ -19,37 +19,55 @@ export class InitializationError extends Error {
   }
 }
 
+export interface AtomicComponentInterface extends ComponentInterface {
+  bindings: Bindings;
+  error?: Error;
+  strings?: Record<string, () => string>;
+}
+
+/**
+ * Utility that automatically fetches the `bindings` from the parent `atomic-search-interface` component. This decorator should be applied to the `initialize` method directly.
+ *
+ * In order for a component using this decorator to render properly, it should have an internal state property using data from the `bindings`. For more information, view the "Utilities" section of the readme.
+ */
 export function Initialization() {
-  return (component: ComponentInterface, initializeMethod: string) => {
+  return (component: AtomicComponentInterface, initializeMethod: string) => {
     const {
       componentWillLoad,
       render,
       componentDidRender,
       componentDidLoad,
-      [initializeMethod]: initialize,
     } = component;
-    const errorProperty = 'error';
-    const contextProperty = 'context';
+    const initialize: () => void = component[initializeMethod];
 
     component.componentWillLoad = function () {
       const element = getElement(this);
       const event = new CustomEvent('atomic/initializeComponent', {
-        detail: (context: InterfaceContext) => {
-          this[contextProperty] = context;
+        detail: (bindings: Bindings) => {
+          this.bindings = bindings;
+
+          // Ensures re-render of localized strings
+          if (this.strings) {
+            this.strings = {...this.strings};
+            this.bindings.i18n.on(
```

Comment:
Should we unsubscribe this event listener when a component unmounts to avoid memory leaks?